sycl/include/sycl/ext/oneapi/matrix/matrix-jit-use.hpp

```diff
@@ -0,0 +1,517 @@
+            Ptr, stride, __spv::MatrixLayout::PackedB,
+            spv_scope_traits<Group>::value);
+    break;
+  }
+#else
+  (void)sg;
+  (void)res;
+  (void)src;
+  (void)stride;
+  (void)MemL;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+}
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_use Use = matrix_use::unnecessary,
+          matrix_layout MatL = matrix_layout::row_major,
+          access::address_space Space>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_store(Group sg,
+                   joint_matrix<T, NumRows, NumCols, Use, MatL, Group> &src,
+                   multi_ptr<T, Space> res, size_t stride, matrix_layout MemL) {
+#ifdef __SYCL_DEVICE_ONLY__
+  T *Ptr = res.get();
+  switch (MemL) {
+  default:
+    assert(false && "Invalid Memory Layout!");
+  case matrix_layout::row_major:
+    __spirv_JointMatrixStoreINTEL<T, NumRows, NumCols,
+                                  spv_matrix_use_traits<Use>::value,
+                                  spv_matrix_layout_traits<MatL>::value>(
+        Ptr, src.spvm, stride, __spv::MatrixLayout::RowMajor,
+        spv_scope_traits<Group>::value);
+    break;
+  case matrix_layout::col_major:
+    __spirv_JointMatrixStoreINTEL<T, NumRows, NumCols,
+                                  spv_matrix_use_traits<Use>::value,
+                                  spv_matrix_layout_traits<MatL>::value>(
+        Ptr, src.spvm, stride, __spv::MatrixLayout::ColumnMajor,
+        spv_scope_traits<Group>::value);
+    break;
+  case matrix_layout::packed_a:
+    __spirv_JointMatrixStoreINTEL<T, NumRows, NumCols,
+                                  spv_matrix_use_traits<Use>::value,
+                                  spv_matrix_layout_traits<MatL>::value>(
+        Ptr, src.spvm, stride, __spv::MatrixLayout::PackedA,
+        spv_scope_traits<Group>::value);
+    break;
+  case matrix_layout::packed_b:
+    __spirv_JointMatrixStoreINTEL<T, NumRows, NumCols,
+                                  spv_matrix_use_traits<Use>::value,
+                                  spv_matrix_layout_traits<MatL>::value>(
+        Ptr, src.spvm, stride, __spv::MatrixLayout::PackedB,
+        spv_scope_traits<Group>::value);
+    break;
+  }
+#else
+  (void)sg;
+  (void)src;
+  (void)res;
+  (void)stride;
+  (void)MemL;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+}
+
+template <typename Group, typename T1, typename T2, typename T3, size_t M,
+          size_t K, size_t N, matrix_use UseA, matrix_use UseB, matrix_use UseC,
+          matrix_layout LayoutA, matrix_layout LayoutB, matrix_layout LayoutC>
+inline __SYCL_ALWAYS_INLINE joint_matrix<T3, M, N, UseC, LayoutC, Group>
+joint_matrix_mad(Group sg, joint_matrix<T1, M, K, UseA, LayoutA, Group> &mA,
```

Comment:
`UseA` is always `matrix_use::matrix_a` for `mA`, `UseB` is always `matrix_use::matrix_b` for `mB` and `UseC` is always `matrix_use::accumulator` for `mC` right? Therefore I see no reason to have these three template parameters in this function at all. If they are removed this function interface will align with the definition in matrix-tensorcores.hpp.

Reply:
AMX store does not restrict it to matrix C.
The use case you are mentioning is true in current hardware. If hardware does element wise operations in the future, we can have these ops applied on A or B and store back to memory. Besides, we already have element wise operations (slicing) added to joint matrix, so this is already possible in software.
In general, load/store are not tied to matrix C.
we needed this use argument for the load operation for the hardware to reason about layout and other things. But we should not restrict load/store API. 
For the specific case where CUDA backend does not support store on matrix A and B, you can have a restriction there in the function, but not on the signature/definition.

Reply:
Sorry, I seem to have missed the comment position.
As for the mad point, I think you are right, we don't need use template parameters on this function
@yubingex007-a11y please confirm